.pipelines/networkobservability/pipeline.yaml

```diff
@@ -0,0 +1,145 @@
+            scriptLocation: "inlineScript"
+            scriptType: "bash"
+            addSpnToEnvironment: true
+            inlineScript: |
+              kubectl apply -f test/integration/manifests/cilium/$(cilium_version)/cilium-config/cilium-config-hubble.yaml
+              kubectl apply -f test/integration/manifests/cilium/$(cilium_version)/cilium-agent/files
+              kubectl apply -f test/integration/manifests/cilium/$(cilium_version)/cilium-operator/files
+              envsubst '${CILIUM_VERSION_TAG},${CILIUM_IMAGE_REGISTRY}' < test/integration/manifests/cilium/$(cilium_version)/cilium-agent/templates/daemonset.tpl | kubectl apply -f -
+              envsubst '${CILIUM_VERSION_TAG},${CILIUM_IMAGE_REGISTRY}' < test/integration/manifests/cilium/$(cilium_version)/cilium-operator/templates/deployment.tpl | kubectl apply -f -
+              # Use different file directories for nightly and current cilium version
+          name: "installCilium"
+          displayName: "Install Cilium on AKS Overlay"
+
+        - script: |
+            echo "Start Azilium E2E Tests on Overlay Cluster"
+            if [ "$CILIUM_VERSION_TAG" = "cilium-nightly-pipeline" ]
+            then
+                CNS=$(CNS_VERSION) IPAM=$(AZURE_IPAM_VERSION) && echo "Running nightly"
+            else
+                CNS=$(make cns-version) IPAM=$(make azure-ipam-version)
+            fi
+            sudo -E env "PATH=$PATH" make test-integration AZURE_IPAM_VERSION=${IPAM} CNS_VERSION=${CNS} INSTALL_CNS=true INSTALL_OVERLAY=true
+          retryCountOnTaskFailure: 3
+          name: "aziliumTest"
+          displayName: "Install Azure-CNS and Run Azilium E2E on AKS Overlay"
+          enabled: true
+
+        - script: |
+              cilium status --wait --wait-duration 5m
+          name: waitforhealthy
+          displayName: "Wait for healthy cilium pods"
+
+        - script: |
+            kubectl wait --for=condition=Ready pods --all --all-namespaces --timeout=5m
+          name: waitforallpodsrunning
+          displayName: "Wait for all pods to be running"
+          retryCountOnTaskFailure: 3
+
+        - script: |
+              echo "Run Cilium Connectivity Tests"
+              cilium status
+              cilium connectivity test --connect-timeout 4s --request-timeout 30s --test '!pod-to-pod-encryption,!node-to-node-encryption'
+          retryCountOnTaskFailure: 3
+          name: "ciliumConnectivityTests"
+          displayName: "Run Cilium Connectivity Tests"
+          enabled: true
+
+        - script: |
+            kubectl apply -f test/integration/manifests/cilium/$(cilium_version)/hubble/hubble-peer-svc.yaml
+            kubectl get pods -Aowide
+            echo "verify Hubble metrics endpoint is usable"
+            go test ./test/integration/networkobservability -tags=networkobservability
+          retryCountOnTaskFailure: 3
+          name: "HubbleConnectivityTests"
+          displayName: "Run Hubble Connectivity Tests"
+          
```

Comment:
If you want you could also add a job here for logs so that way you could look at cilium files from the nightly run. https://github.com/Azure/azure-container-networking/blob/5c1fe966129a287533cc90f596ce19c19aff65ff/.pipelines/cni/cilium/nightly-release-test.yml#L107-L118

Reply:
Sounds reasonable and useful for troubleshooting.